llama_stack/providers/remote/inference/groq/groq.py

```diff
@@ -0,0 +1,142 @@
+import warnings
+from typing import AsyncIterator, List, Optional, Union, AsyncGenerator
+from llama_models.llama3.api.datatypes import (
+    InterleavedTextMedia,
+    Message,
+    ToolChoice,
+    ToolDefinition,
+    ToolPromptFormat,
+)
+from llama_models.datatypes import SamplingParams
+from llama_stack.apis.inference import (
+    ChatCompletionRequest,
+    ChatCompletionResponse,
+    ChatCompletionResponseEvent,
+    ChatCompletionResponseStreamChunk,
+    ChatCompletionResponseEventType,
+    CompletionResponse,
+    CompletionMessage,
+    CompletionResponseStreamChunk,
+    EmbeddingsResponse,
+    Inference,
+    LogProbConfig,
+    ResponseFormat,
+)
+from llama_stack.providers.utils.inference.model_registry import (
+    build_model_alias,
+    ModelRegistryHelper,
+)
+from llama_stack.providers.remote.inference.groq.config import GroqConfig
+from llama_models.sku_list import CoreModelId
+from llama_models.llama3.api.datatypes import StopReason
+from groq import Groq
+
+_MODEL_ALIASES = [
+    build_model_alias(
+        "llama3-8b-8192",
+        CoreModelId.llama3_2_3b_instruct.value,
+    ),
+]
+
+class GroqInferenceAdapter(Inference, ModelRegistryHelper):
+    client: Groq
+    def __init__(self, config: GroqConfig) -> None:
+        ModelRegistryHelper.__init__(self, model_aliases=_MODEL_ALIASES)
+        self._client = Groq(api_key=config.api_key)
+
+    def completion(
+        self,
+        model_id: str,
+        content: InterleavedTextMedia,
+        sampling_params: Optional[SamplingParams] = SamplingParams(),
+        response_format: Optional[ResponseFormat] = None,
+        stream: Optional[bool] = False,
+        logprobs: Optional[LogProbConfig] = None,
+    ) -> Union[CompletionResponse, AsyncIterator[CompletionResponseStreamChunk]]:
+        raise NotImplementedError()
+
+    async def embeddings(
+        self,
+        model_id: str,
+        contents: List[InterleavedTextMedia],
+    ) -> EmbeddingsResponse:
+        raise NotImplementedError()
+
+    async def chat_completion(
+        self,
+        model_id: str,
+        messages: List[Message],
+        sampling_params: Optional[SamplingParams] = SamplingParams(),
+        response_format: Optional[ResponseFormat] = None,
+        tools: Optional[List[ToolDefinition]] = None,
+        tool_choice: Optional[ToolChoice] = ToolChoice.auto,
+        tool_prompt_format: Optional[
+            ToolPromptFormat
+        ] = None,  # API default is ToolPromptFormat.json, we default to None to detect user input
+        stream: Optional[bool] = False,
+        logprobs: Optional[LogProbConfig] = None,
+    ) -> Union[
+        ChatCompletionResponse, AsyncGenerator
+    ]:
+        if logprobs:
+            warnings.warn("logprobs are not supported yet")
+        
+        if response_format:
+            warnings.warn("response_format is not supported yet")
```

Comment:
JSON mode still in beta for Groq:

<img width="764" alt="image" src="https://github.com/user-attachments/assets/1504963c-0d28-47f5-8a09-777980b2e697">

https://console.groq.com/docs/text-chat#json-mode-object-object